dipy/viz/horizon/tab/base.py

```diff
@@ -103,16 +106,29 @@ def __init__(self, tabs, win_size, synchronize_slices=False):
 
         self._tab_ui.on_change = self._tab_selected
 
-        self.tab_changed = lambda actors: None
-
         for tab_id, tab in enumerate(tabs):
-            self._tab_ui.tabs[tab_id].title = ' ' + tab.name
             self._tab_ui.tabs[tab_id].title_font_size = 18
             tab.build(tab_id, self._tab_ui)
             if tab.__class__.__name__ == 'SlicesTab':
                 tab.on_slice_change = self.synchronize_slices
                 self._render_tab_elements(tab_id, tab.elements)
 
+        self._scene.add(self._tab_ui)
+
+        for tab_id, tab in enumerate(tabs):
+            self._handle_title_overflow(
+                tab.name,
+                self._tab_ui.tabs[tab_id].text_block
+            )
+
+    def _handle_title_overflow(self, title_text, title_block):
+        tab_text = remove_extension(title_text, 'nii')
```

Comment:
Not sure you need a function for that....    

`tab_text = title_text.split('.', 1)[0]` should just be enough.....

in `pathlib` module, you also have `suffixes()`